includes/payment-methods/class-wc-stripe-upe-payment-method.php

```diff
@@ -284,8 +284,9 @@ public function is_enabled_at_checkout( $order_id = null, $account_domestic_curr
 		}
 
 		// If cart or order contains pre-order, enable payment method if it's reusable.
+		// BLIK supports pre-order when product is charged upfront. We're handling availability in WC_Stripe_UPE_Payment_Method_BLIK.
 		if ( $this->is_pre_order_item_in_cart() || ( ! empty( $order_id ) && $this->has_pre_order( $order_id ) ) ) {
-			return $this->is_reusable();
+			return $this->is_reusable() || WC_Stripe_Payment_Methods::BLIK === $this->stripe_id;
```

Comment:
I believe the logic we have introduced for making PMs available for pre-orders with charge upfront vs. charge upon release is not ideal.

I think all PMs can handle charge upfront, so we could move the logic introduced in `maybe_hide_bacs_payment_gateway` and `maybe_hide_blik` here, and then have some other way of defining if the PM supports charge upon release, either with `$this->is_reusable()`, or some new property like `$this->supports_charge_upon_release()` which defaults to `$this->is_reusable` for all other PMs but BLIK.

I know this isn't in the scope for this PR, so maybe we should simply add a TODO comment here with reference to this comment or a new issue in Linear if you agree.